pkg/ebpf/bpf_client.go

```diff
@@ -438,15 +442,28 @@ func (l *bpfClient) GetEgressProgToPodsMap() *sync.Map {
 	return l.EgressProgToPodsMap
 }
 
-func (l *bpfClient) AttacheBPFProbes(pod types.NamespacedName, podIdentifier string, ingress bool, egress bool) error {
+func (l *bpfClient) AttacheBPFProbes(pod types.NamespacedName, podIdentifier string) error {
+	podNamespacedName := utils.GetPodNamespacedName(pod.Name, pod.Namespace)
+	// In strict mode two go routines can try to attach the probes at the same time
+	// Locking will help updating all the datastructures correctly
+	value, _ := l.PodToAttachProbesLock.LoadOrStore(podNamespacedName, &sync.Mutex{})
+	attachProbesLock := value.(*sync.Mutex)
+	attachProbesLock.Lock()
+	l.logger.Info("Got the attatchProbesLock for", "pod", pod.Name, " in namespace", pod.Namespace)
```

Comment:
spelling mistake here. attatch

Reply:
more importantly, I don't see this the logs output pasted (in the correct lock version).

Reply:
Will fix this

Reply:
This is the line where we are getting the lock in the logs posted in PR description 

{"level":"info","ts":"2024-10-23T07:07:02.756Z","logger":"ebpf-client","caller":"ebpf/bpf_client.go:452","msg":"Got the lock and going ahead with attach probes"}

Modified log wording after collecting logs

Reply:
fixed typo